lib/IRGen/GenClass.cpp

```diff
@@ -726,6 +726,22 @@ void irgen::emitPartialClassDeallocation(IRGenFunction &IGF,
                                          llvm::Value *metadataValue) {
   auto *theClass = selfType.getClassOrBoundGenericClass();
 
+  // Foreign classes should not be freed by sending -release.
+  // They should also probably not be freed with object_dispose(),
+  // either.
+  //
+  // However, in practice, the only time we should try to free an
+  // instance of a foreign class here is inside an initializer
+  // delegating to a factory initializer. In this case, the object
+  // was allocated with +allocWithZone:, so calling object_dispose()
+  // should be OK.
+  if (theClass->getForeignClassKind() == ClassDecl::ForeignKind::RuntimeOnly) {
```

Comment:
This is the "Jordan hates this patch" as mentioned in the commit message. It's not really the correct fix for deallocating a partially-allocated foreign class, but in practice it'll do the right thing and {{release}} won't, because some of them are in a state where {{dealloc}} shouldn't be called.